bin/storm.py

```diff
@@ -296,787 +251,1090 @@ def exec_storm_class(klass, jvmtype="-server", jvmopts=[], extrajars=[], args=[]
+        "get-errors", help="""Get the latest error from the running topology. The returned result contains
+    the key value pairs for component-name and component-error for the components in error.
+    The result is returned in json format.""", formatter_class=SortingHelpFormatter
+    )
+
+    sub_parser.add_argument("topology-name")
+
+    sub_parser.set_defaults(func=get_errors)
+    add_common_options(sub_parser)
+
+
+def initialize_healthcheck_subcommand(subparsers):
+    sub_parser = subparsers.add_parser(
+        "node-health-check", help="""Run health checks on the local supervisor.""", formatter_class=SortingHelpFormatter
+    )
+
+    sub_parser.set_defaults(func=healthcheck)
+    add_common_options(sub_parser)
+
+
+def initialize_kill_workers_subcommand(subparsers):
+    sub_parser = subparsers.add_parser(
+        "kill_workers", help="""Kill the workers running on this supervisor. This command should be run
+    on a supervisor node. If the cluster is running in secure mode, then user needs
+    to have admin rights on the node to be able to successfully kill all workers.""", formatter_class=SortingHelpFormatter
+    )
+
+    sub_parser.set_defaults(func=kill_workers)
+    add_common_options(sub_parser)
+
+
+def initialize_admin_subcommand(subparsers):
+    sub_parser = subparsers.add_parser("admin", help="""The storm admin command provides access to several operations that can help
+    an administrator debug or fix a cluster.""", formatter_class=SortingHelpFormatter)
+    sub_sub_parsers = sub_parser.add_subparsers()
+
+    remove_sub_sub_parser = sub_sub_parsers.add_parser(
+        "remove_corrupt_topologies", help="""This command should be run on a nimbus node as
+    the same user nimbus runs as.  It will go directly to zookeeper + blobstore
+    and find topologies that appear to be corrupted because of missing blobs.
+    It will kill those topologies.""", formatter_class=SortingHelpFormatter
+    )
+
+    add_common_options(remove_sub_sub_parser)
 
-    If for some reason you need to have the full storm classpath, not just the one for the worker you may include the command line option `--storm-server-classpath`.  Please be careful because this will add things to the classpath that will not be on the worker classpath and could result in the worker not running.
+    zk_cli_parser = sub_sub_parsers.add_parser(
+        "zk_cli", help="""This command will launch a zookeeper cli pointing to the
+    storm zookeeper instance logged in as the nimbus user.  It should be run on
+    a nimbus server as the user nimbus runs as.""", formatter_class=SortingHelpFormatter
+    )
+
+    zk_cli_parser.add_argument(
+        "-s", "--server", default=None, help="""Set the connection string to use,
+        defaults to storm connection string"""
+    )
+
+    zk_cli_parser.add_argument(
+        "-t", "--time-out", default=None, help="""Set the timeout to use, defaults to storm
+            zookeeper timeout."""
+    )
```

Comment:
* `type=check_positive`?
* Does this have to be positive? What happens on `0`?
* What are the units?

Reply:
This has been addressed.